test/reactors/V3DutchOrderReactor.t.sol

```diff
@@ -0,0 +1,1005 @@
+
+        SignedOrder memory order = generateOrder(
+            TestDutchOrderSpec({
+                currentBlock: block.number,
+                startBlock: startBlock,
+                deadline: deadline,
+                input: V3DutchInput(
+                    tokenIn,
+                    inputAmount,
+                    CurveBuilder.singlePointCurve(1000, 0 - int256(inputAmount * 10 / 100)),
+                    inputAmount * 110 / 100
+                ),
+                outputs: OutputsBuilder.singleV3Dutch(
+                    address(tokenOut),
+                    outputAmount,
+                    CurveBuilder.singlePointCurve(1000, int256(inputAmount * 10 / 100)),
+                    address(swapper)
+                )
+            })
+        );
+        uint256 swapperInputBalanceStart = tokenIn.balanceOf(address(swapper));
+        uint256 swapperOutputBalanceStart = tokenOut.balanceOf(address(swapper));
+        vm.expectEmit(false, true, true, false, address(reactor));
+        emit Fill(keccak256("not checked"), address(fillContract), swapper, 0);
+        vm.roll(startBlock + 500);
+        fillContract.execute(order);
+        uint256 swapperInputBalanceEnd = tokenIn.balanceOf(address(swapper));
+        uint256 swapperOutputBalanceEnd = tokenOut.balanceOf(address(swapper));
+        assertEq(swapperInputBalanceStart - swapperInputBalanceEnd, inputAmount * 105 / 100);
+        assertEq(swapperOutputBalanceEnd - swapperOutputBalanceStart, outputAmount * 95 / 100);
+    }
+
+    function testV3ExecuteInputAndOutputFullDecay() public {
+        uint256 inputAmount = 1 ether;
+        uint256 outputAmount = 1 ether;
+        uint256 startBlock = block.number;
+        uint256 deadline = block.timestamp + 1000;
+        tokenOut.mint(address(fillContract), uint256(outputAmount) * 100);
+
+        SignedOrder memory order = generateOrder(
+            TestDutchOrderSpec({
+                currentBlock: block.number,
+                startBlock: startBlock,
+                deadline: deadline,
+                input: V3DutchInput(
+                    tokenIn,
+                    inputAmount,
+                    CurveBuilder.singlePointCurve(1000, 0 - int256(inputAmount * 10 / 100)),
+                    inputAmount * 110 / 100
+                ),
+                outputs: OutputsBuilder.singleV3Dutch(
+                    address(tokenOut),
+                    outputAmount,
+                    CurveBuilder.singlePointCurve(1000, int256(inputAmount * 10 / 100)),
+                    address(swapper)
+                )
+            })
+        );
+        uint256 swapperInputBalanceStart = tokenIn.balanceOf(address(swapper));
+        uint256 swapperOutputBalanceStart = tokenOut.balanceOf(address(swapper));
+        vm.expectEmit(false, true, true, false, address(reactor));
+        emit Fill(keccak256("not checked"), address(fillContract), swapper, 0);
```

Comment:
can we not check against the local order.hash() here?

Reply:
I believe this is saying that the `vm.expectEmit()` doesn't check the value here (I took this from the V2 tests).